### PR TITLE
Implemented range_replace

### PR DESCRIPTION
**Pull request by shamis**

### CODE REVIEW

File: src/imem_dal_skvh.erl [hunk 1]
```diff
@@ -1058,6 +1059,32 @@ read_with_limit(Cmd, SkvhCtx, Item, MatchFunction, Limit) ->
...
+    range_replace(User, atom_table_name(Channel), CKey1, CKey2, KVTable);
+range_replace(User, TableName, CKey1, CKey2, KVTable) ->
+    CKey1E = term_key_to_binterm(CKey1),
+    CKey2E = term_key_to_binterm(CKey2),
+    KVTableE = [{term_key_to_binterm(Key), Value} || {Key, Value} <- KVTable],
+    ReplaceFun = fun() ->
+        RemainingKV = range_replace_internal(User, TableName, CKey1E, CKey2E, KVTableE),
+        [imem_meta:merge(TableName, #skvhTable{ckey = Key, cvalue = Value}, User) || {Key, Value} <- RemainingKV]
+    end,
+    imem_meta:transaction(ReplaceFun).
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Use https://github.com/K2InformaticsGmbH/imem/blob/master/src/imem_meta.erl#L2891
```erlang
imem_meta:transaction(fun range_replace_internal/5, [User, TableName, CKey1E, CKey2E, KVTableE])
```

File: src/imem_dal_skvh.erl [hunk 2]
```diff
@@ -1058,6 +1059,32 @@ read_with_limit(Cmd, SkvhCtx, Item, MatchFunction, Limit) ->
...
+        NextKey -> 
+            case lists:keytake(NextKey, 1, KVTable) of
+                false -> 
+                    imem_meta:delete(TableName, NextKey),
+                    range_replace_internal(User, TableName, NextKey, EndKey, KVTable);
+                {value, {Key, Value}, NewKVTable} -> 
+                    imem_meta:merge(TableName, #skvhTable{ckey = Key, cvalue = Value}, User),
+                    range_replace_internal(User, TableName, NextKey, EndKey, NewKVTable)
+            end
+    end.
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Move `[imem_meta:merge(TableName, #skvhTable{ckey = Key, cvalue = Value}, User) || {Key, Value} <- RemainingKV]` into `range_replace_internal`

File: src/imem_dal_skvh.erl [hunk 3]
```diff
@@ -1058,6 +1059,32 @@ read_with_limit(Cmd, SkvhCtx, Item, MatchFunction, Limit) ->
 		true ->					project_result(Cmd, L, Item)
 	end.
 
+range_replace(User, Channel, CKey1, CKey2, KVTable) when is_binary(Channel) ->
+    range_replace(User, atom_table_name(Channel), CKey1, CKey2, KVTable);
+range_replace(User, TableName, CKey1, CKey2, KVTable) ->
+    CKey1E = term_key_to_binterm(CKey1),
+    CKey2E = term_key_to_binterm(CKey2),
+    KVTableE = [{term_key_to_binterm(Key), Value} || {Key, Value} <- KVTable],
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Rename `KVTable` to `KVTuples`

**Comment by shamis** [comment]
@c-bik Have made the changes please verify

**Comment by c-bik** [comment]
@stoch Please confirm if this API is still required.

**Pull request closed without merging**